src/service/engine/index.ts

```diff
@@ -1,67 +1,79 @@
-import { ACCOUNT_ENGINE, ACCOUNT_ENGINES, ResultsPage } from "../../common/api";
+import {
+  AuthenticationError,
+  ConnectionError,
+  DeprecationError
+} from "../../common/errors";
+import { Connection } from "../../connection";
 import { Context } from "../../types";
 import { EngineModel } from "./model";
-import { ID, Engine } from "./types";
+import { EngineStatusSummary } from "./types";
 
 export class EngineService {
   private context: Context;
+  connection!: Connection;
 
   constructor(context: Context) {
     this.context = context;
   }
 
-  private async getEngineId(engineName: string): Promise<ID> {
-    const { apiEndpoint, httpClient } = this.context;
-    const accountId = this.context.resourceManager.account.id;
-    const queryParams = new URLSearchParams({ engine_name: engineName });
-    const url = `${apiEndpoint}/${ACCOUNT_ENGINES(
-      accountId
-    )}:getIdByName?${queryParams}`;
-    const data = await httpClient
-      .request<{ engine_id: ID }>("GET", url)
-      .ready();
-    return data.engine_id;
+  async getById(engineId: string) {
+    throw new DeprecationError({
+      message: "Can't call getById as engine IDs are deprecated"
+    });
   }
 
-  async getById(engineId: string): Promise<EngineModel> {
-    const { apiEndpoint, httpClient } = this.context;
-    const accountId = this.context.resourceManager.account.id;
-    const url = `${apiEndpoint}/${ACCOUNT_ENGINE(accountId, engineId)}`;
-    const data = await httpClient
-      .request<{ engine: Engine }>("GET", url)
-      .ready();
-    return new EngineModel(this.context, data.engine);
+  private throwErrorIfNoConnection() {
+    if (typeof this.connection == "undefined") {
+      throw new AuthenticationError({
+        message:
+          "Can't execute a resource manager operation. Did you run authenticate()?"
+      });
+    }
   }
 
   async getByName(engineName: string): Promise<EngineModel> {
-    const { engine_id } = await this.getEngineId(engineName);
-    const engine = await this.getById(engine_id);
-    return new EngineModel(this.context, engine);
+    this.throwErrorIfNoConnection();
+    const query =
+      "SELECT engine_name, url, status FROM information_schema.engines " +
+      `WHERE engine_name='${engineName}'`;
+    const statement = await this.connection.execute(query);
+    const { data } = await statement.fetchResult();
+    if (data.length == 0) {
+      throw new ConnectionError({
+        message: `Engine ${engineName} not found or is not accessbile`
+      });
+    }
+    const firstRow = data[0] as unknown[];
+    const status: EngineStatusSummary = firstRow[2] as EngineStatusSummary;
+    const engine = {
+      name: firstRow[0] as string,
```

Comment:
hard to follow with array indexes. You can use something like `const [name, endpoint, status] = row`